hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java

```diff
@@ -279,17 +291,20 @@ public boolean isRunning() {
         if (isEmpty(parentId)) {
             return Collections.emptyList();
         }
-        return qResource.from(Resource.class)
+        return qResource.from(IspnResource.class)
                 .having("parentId").equal(parentId)
                 .build()
-                .list();
+                .list()
+                .stream()
+                .map(r -> ((IspnResource) r).getResource())
+                .collect(Collectors.toList());
     }
 
     private ResourceType getNullableResourceType(String typeId) {
         if (isEmpty(typeId)) {
             return null;
         }
-        return (ResourceType) resourceType.get(typeId);
+        return ((IspnResourceType) resourceType.get(typeId)).getResourceType();
```

Comment:
return value here can be null if the type doesn't exist, so it's missing a null-check

Reply:
+1 good catch, I missed this one